week3/homework/src/actionsAndStaff/writeFindCheckError.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const fs = require('fs');
+
+const readAndParse = require('./readTodo');
+
+const LIST_FILE_PATH = __dirname + '/list.json';
+
+class ErrorWithStatus extends Error {
+  constructor(errorMessage, statusCode) {
+    super(errorMessage);
+    this.name = 'ErrorWithStatus';
+    this.statusCode = statusCode;
+  }
+}
+
+async function write(jsonData) {
+  try {
+    await fs.writeFile(LIST_FILE_PATH, JSON.stringify(jsonData, null, 2), error => {
+      if (error) throw error;
```

Comment:
I have not tested this but I think it will not work. You need to promisify writeFile first
```
await new Promise((resolve,reject) => fs.writeFile("foo.txt", "example", error => {
       if ( error) reject(error);
       resolve();
}
```
Or use https://www.npmjs.com/package/promise-fs

Reply:
I have tested as a whole and everything was working. Perhaps something that I could not think of while testing would cause a problem here. I will use `promisify` anyway, so that the code seems more consistent since I use promise for `fs.readFile`.